core/src/main/java/com/solarwinds/joboe/core/profiler/Profiler.java

```diff
@@ -307,19 +315,17 @@ private Profile() {
          * @return
          */
         private Metadata createProfileSpanEntry(Metadata parentMetadata) {
-            Metadata entryMetadata = new Metadata(parentMetadata);
+            entryMetadata = new Metadata(parentMetadata);
 
-            Event snapshotEntry = Context.createEventWithContext(entryMetadata, false);
+            snapshotEntry = Context.createEventWithContext(entryMetadata, false);
+            snapshotEntry.setTimestamp(TimeUtils.getTimestampMicroSeconds());
             snapshotEntry.addInfo("Label", "entry",
                                   "Spec", "profiling",
                                   "Language", "java",
                                   "Interval", (int) interval,
                                   "SpanRef", parentMetadata.opHexString());
 
-
-            snapshotEntry.report(entryMetadata, Profiler.reporter);
-
-            return entryMetadata;
+            return snapshotEntry.getMetadata();
```

Comment:
Return the entry event metadata because it's not reported immediately. This helps maintain the hierarchy because parent id is only updated after reporting. See [here](https://github.com/solarwinds/joboe/blob/689549b30dc868a8b74ac8220bac360efce43f8f/core/src/main/java/com/solarwinds/joboe/core/EventImpl.java#L265)